### PR TITLE
Add job annotations

### PR DESCRIPTION
**Pull request by onyxraven**
This adds the ability to add annotations to brigade job pods from the brigade.js brigade-worker.

We ran across the need for this to be able to run workloads in ec2 and take advantage of [kube2iam](https://github.com/jtblin/kube2iam#kubernetes-annotation), which needs an annotation.

I tested this in minikube, adding a dummy annotation to my brigade pod:

```javascript
  // Create a new job
  var node = new Job("copy-staging-db");

  // snip

  node.tasks = [
    'env',
    'hostname'
  ];

  node.annotations = {
    'test/annotation': "annotations/go/here"
  };

  // We're done configuring, so we run the job
  node.run(); 
```
and saw the annotation applied on the pod via the dashboard
screen shot 2018-04-13 at 8 30 37 am

I'm not currently sure this adversely affects anything else - it doesnt seem like it does, the tests were testing to ensure that annotations were not set except for the init container, and those just changed to asserting an empty object.

Happy to answer any other questions.

### CODE REVIEW

**Comment by onyxraven** [comment]
Yep, we'll look into it. 

for the CLA, I'm awaiting corporate approval (I can't imagine there will be a problem).

**Comment by technosophos** [comment]
I am dying with laughter (and/or embarrassment) right now. I just clicked on your user name, and saw that it was you, Justin!

**Comment by jdotjdot** [comment]
I think it's worth noting that bit by bit I'm observing Javascript features being added that just replicate (or half-replicate) all of the features that already exist in the Kubernetes YAML.

Would it be worth just adding an option that lets Brigade simply grab a YAML declaration and run it?  The current `Job` structure can be kept to make it easier for people who don't know (or what to learn) complex Kubernetes set-ups, but for those who want that control or have pre-existing container configs they want to run, it could make all of this a lot easier.

**Comment by onyxraven** [comment]
@jdotjdot that's interesting, but I wonder if that exposes too much of the internals of how the job actually runs - for example, if it were to change from a Pod to a k8s Job.  In this case, the args we're specifying (args, tasks, annotations) are common among the final outputs.

@technosophos What are you looking for as far as real testing? In minikube, it applied the annotations I supplied, or are you looking to ensure it all works on a full k8s install?

**Comment by technosophos** [comment]
@jdotjdot we're trying to be careful about how much of Kubernetes to expose. The main reason is that we need to be able to effectively manage the job objects, and that means doing a certain amount of stuff "for you", and making it really hard to break architectural features. And at the same time, we're trying to make it easy enough to use that a user needn't learn the details of Kubernetes to write a basic script.

That said, we're also open to exposing deeper parts of Kubernetes as we discover the need (provided doing so is "safe"). This is why I am open to adding many of the things you and others have been asking for.

In short, we're just shooting for a reasonable abstraction that hides our own implementation details, protects certain assumptions, and still meets the needs of developers.

@onyxraven I think I'll give this one a spin on my minikube cluster tomorrow. I was assuming that this needed some AWS-specific testing, but it looks like that's actually just incidental to the feature.

**Comment by onyxraven** [comment]
@technosophos signed. Let me know if theres anything I can help with testing

**Comment by blimmer** [comment]
Btw this fixes #421

File: brigade-worker/test/job.ts [hunk 1]
```diff
@@ -134,5 +134,15 @@ describe("job", function() {
         assert.isFalse(j.storage.enabled);
       });
     });
+    describe("#annotations", function() {
+      beforeEach(function() {
+        j = new mock.MockJob("my-job");
+      });
+      it("is a empty list that can be written", function() {
```
Comment (technosophos) [inline brigade-worker/test/job.ts]: Very minor non-blocking nit: "is an empty list"

Comment (onyxraven) [reply brigade-worker/test/job.ts]: lol, i can fix that up.

**Comment by blimmer** [comment]
Thanks @onyxraven , this will help to unblock our kube2iam usage :-)
